packages/yarnpkg-core/sources/Configuration.ts

```diff
@@ -1330,16 +1337,24 @@ export class Configuration {
     }
   }
 
-  useWithSource(source: string, data: {[key: string]: unknown}, folder: PortablePath, opts?: {strict?: boolean, overwrite?: boolean}) {
+  private getPackageEnv(packageCwd: PortablePath | null) {
+    if (!packageCwd) return {};
+    const envAbsolutePath = ppath.resolve(packageCwd, `.env` as PortablePath);
+    if (!xfs.existsSync(envAbsolutePath)) return {};
+    const envFile = xfs.readFileSync(envAbsolutePath, `utf8`);
+    return parseEnv(envFile);
+  }
```

Comment:
```suggestion
  private async findPackageEnv(packageCwd: PortablePath | null) {
    if (!packageCwd)
      return {};

    const envAbsolutePath = ppath.resolve(packageCwd, `.env` as PortablePath);

    let envFile: string;
    try {
      return parseEnv(await xfs.readFilePromise(envAbsolutePath, `utf8`));
    } catch (err: any) {
      if (err.code === `ENOENT`) {
        return {};
      } else {
        throw err;
      }
    }
  }
```

Reply:
Great~~ I've added it to this commit https://github.com/yarnpkg/berry/pull/4835/commits/d413eff313d84f2e8b453f40a5f36296876ff08b